thorlcr/activities/hashdistrib/thhashdistribslave.cpp

```diff
@@ -3737,36 +3737,224 @@ class HashJoinSlaveActivity : public CSlaveActivity, implements IStopInput
 
 //===========================================================================
 
-RowAggregator *mergeLocalAggs(Owned<IHashDistributor> &distributor, CActivityBase &activity, IHThorRowAggregator &helper, IHThorHashAggregateExtra &helperExtra, RowAggregator *localAggTable, mptag_t mptag, bool ordered)
+class CAggregateHT : public CSimpleInterfaceOf<IAggregateTable>
+{
+    struct HTEntry
+    {
+        void *row;
+        unsigned hash;
+        size32_t size;
+    };
+    CActivityBase &activity;
+    Owned<IEngineRowAllocator> rowAllocator;
+    IHash * hasher;
+    IHash * elementHasher;
+    ICompare * comparer;
+    ICompare * elementComparer;
+    IHThorRowAggregator & helper;
+
+    unsigned htn;
+    unsigned n;
+    HTEntry *table;
+    unsigned iPos = 0;
+
+    void expand()
+    {
+        HTEntry *t = table+htn; // more interesting going backwards
```

Comment:
Not sure, it was originally a line+comment copied from CMinHashTable. I should have removed here though.